ksonnet-gen/astext/astext.go

```diff
@@ -34,7 +34,7 @@ type Object struct {
 
 var (
 	// reFieldStr matches a field id that should be enclosed in quotes.
-	reFieldStr = regexp.MustCompile(`^[_A-Za-z]+[_A-Za-z0-9\-]*$`)
+	reFieldStr = regexp.MustCompile(`^([_A-Za-z0-9\.]?[A-Za-z0-9\-_\.]+(\.[A-Za-z0-9\-_]+)*)?$`)
```

Comment:
This seems to be more restrictive than the jsonnet spec for strings, but more inclusive than the spec for identifiers. If these fall under the rules of strings, is there any need to have this second pattern, or can !match(reField) be enough to quote below?